Fix sample count for phylo trees

ttung opened the pull request:
### Description
Sample counts should reflect what's in the phylo tree.

The previous query was not working because sqlalchemy was creating an implicit join to the PhyloRun table in the inner query.  What we want is a correlated query, where the PhyloRun is coming from the outer query.  Use an alias to explicitly name which PhyloRun we want.

Depends on #326 

#### Issue
[ch135382](https://app.clubhouse.io/genepi/story/135382/number-of-samples-in-tree-table-view-seems-off)

### Test plan
Added a test to test the phylo_tree endpoint, which verifies the sample counts.

phoenixAja commented on src/backend/aspen/app/views/tests/test_phylo_tree_view.py:
```diff
@@ -0,0 +1,53 @@
+    samples = [
+        sample_factory(
+            group,
+            public_identifier=f"public_identifier_%{ix}",
+            private_identifier=f"private_identifier_%{ix}",
```
are you mixing up two different types of string formatting here?

ttung replied:
Yes, there should just be no `%`

ttung replied:
good catch!

Commit:
Fix sample count for phylo trees

### Description
Sample counts should reflect what's in the phylo tree.

The previous query was not working because sqlalchemy was creating an implicit join to the PhyloRun table in the inner query.  What we want is a correlated query, where the PhyloRun is coming from the outer query.  Use an alias to explicitly name which PhyloRun we want.

#### Issue
[ch135382](https://app.clubhouse.io/genepi/story/135382/number-of-samples-in-tree-table-view-seems-off)

### Test plan
Added a test to test the phylo_tree endpoint, which verifies the sample counts.
